tests/functionaltests/Tests/UIKitTests/UIButtonTests.mm

```diff
@@ -118,68 +117,1144 @@ - (void)dealloc {
+
+                // Validation
+                EXPECT_EQ(width, 30.0f);
+                uxEvent->Set();
+            });
+            xamlHeightSubscriber->Set(xamlElement, [WXCImage heightProperty], ^(WXDependencyObject* sender, WXDependencyProperty* dp) {
+                double height = DoubleFromPropertyValue([sender getValue:dp]);
+
+                // Validation
+                EXPECT_EQ((int)height, 16);
+                uxEvent->Set();
+            });
+            xamlCanvasLeftSubscriber->Set(xamlElement, [WXCCanvas leftProperty], ^(WXDependencyObject* sender, WXDependencyProperty* dp) {
+                double left = DoubleFromPropertyValue([sender getValue:dp]);
+
+                // Validation
+                EXPECT_EQ(left, 110.0f);
+                uxEvent->Set();
+            });
+            xamlCanvasTopSubscriber->Set(xamlElement, [WXCCanvas topProperty], ^(WXDependencyObject* sender, WXDependencyProperty* dp) {
+                double top = DoubleFromPropertyValue([sender getValue:dp]);
+
+                // Validation
+                EXPECT_EQ(top, 91.0f);
+                uxEvent->Set();
+            });
+
+            // Action
+            buttonToTest.contentEdgeInsets = UIEdgeInsetsMake(0.0f, 60.0f, 0.0f, 60.0f);
+        });
+
+        ASSERT_TRUE_MSG(uxEvent->Wait(c_testTimeoutInSec, 4 /* signal count */), "FAILED: Waiting for property changed events timed out!");
+    }
+
+    TEST_METHOD(UIButton_TitleEdgeInsets) {
+        StrongId<UIButtonWithControlsViewController> buttonVC;
+        buttonVC.attach([[UIButtonWithControlsViewController alloc] init]);
+        UXTestAPI::ViewControllerPresenter testHelper(buttonVC);
+
+        UIButton* buttonToTest = [buttonVC defaultButton];
+
+        __block auto uxEvent = UXEvent::CreateAuto();
+        __block auto xamlWidthSubscriber = std::make_shared<XamlEventSubscription>();
+        __block auto xamlHeightSubscriber = std::make_shared<XamlEventSubscription>();
+        __block auto xamlCanvasLeftSubscriber = std::make_shared<XamlEventSubscription>();
+        __block auto xamlCanvasTopSubscriber = std::make_shared<XamlEventSubscription>();
+
+        // Extract UIButton.titleLabel control to verify its visual state
+        WXFrameworkElement* xamlElement = [buttonToTest.titleLabel xamlElement];
+        ASSERT_OBJCNE(xamlElement, nil);
+
+        dispatch_sync(dispatch_get_main_queue(), ^{
+            [buttonVC textTitleStateField].text = @"N;";
+            [buttonVC textTitle].text = @"Normal";
+            [buttonVC textTitleColorStateField].text = @"N;";
+            [buttonVC textTitleColor].text = @"redColor";
+
+            // Register RAII event subscription handler
+            xamlWidthSubscriber->Set(xamlElement, [WXCImage widthProperty], ^(WXDependencyObject* sender, WXDependencyProperty* dp) {
+                double width = DoubleFromPropertyValue([sender getValue:dp]);
+
+                // Validation
+                EXPECT_EQ(width, 40.0f);
```

Comment:
we should make that calculation (contract) clear. otherwise, if the test failed with a value, let us say 41. it becomes difficult to track down it is our contract failed vs impl regression?

---
In reply to: [100697740](https://github.com/Microsoft/WinObjC/pull/1947#discussion_r100697740) [](ancestors = 100697740,100681068)